### PR TITLE
feat(providers): add Ory Hydra provider

### PR DESCRIPTION
**Pull request by tomcastro**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## ☕️ Reasoning

Adding Ory Hydra as an OAuth2 provider for logging in.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by vinckr** [comment]
Hello reviewers,
anything I can do to move this PR forward? 
Would love to add an official example with NextAuth to our documentation!

File: packages/next-auth/provider-logos/ory-dark.svg
Comment (ThangHuuVu) [inline packages/next-auth/provider-logos/ory-dark.svg:1]: please help remove this file, it's not needed 🙏

File: packages/next-auth/provider-logos/ory.svg
Comment (ThangHuuVu) [inline packages/next-auth/provider-logos/ory.svg:1]: please help remove this file, it's not needed 🙏

File: packages/core/src/providers/ory-hydra.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * :::
+ */
+export default function OryHydra<P extends OryHydraProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "hydra",
+    name: "Hydra",
+    type: "oauth",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/ory-hydra.ts:80]: We can put this provider as `oidc` type. Then, we can remove the `wellKnown` options below, as its automatically added by `next-auth` core.

File: packages/core/src/providers/ory-hydra.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    authorization: { params: { scope: "openid offline_access" } },
+    wellKnown: `${options.issuer}/.well-known/openid-configuration`,
+    checks: ["pkce", "state"],
+    profile(profile) {
+      return {
+        id: profile.sub,
```
Comment (ThangHuuVu) [inline packages/core/src/providers/ory-hydra.ts:86]: why only return id? we normally return name, email and picture, too.

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]

**Comment by ndom91** [comment]
@tomcastro hey if you'd still like this to be merged can you please remove the `next-auth/provider-logos/*` files as well as take a look at the merge conflicts and remove the `pnpm-lock.yaml` changes? Thanks!

**Pull request merged**